src/test/fuzz/util.h

```diff
@@ -534,36 +534,37 @@ class FuzzedSock : public Sock
 {
     FuzzedDataProvider& m_fuzzed_data_provider;
 
+    /**
+     * Data to return when `MSG_PEEK` is used as a `Recv()` flag.
+     * If `MSG_PEEK` is used, then our `Recv()` returns some random data as usual, but on the next
+     * `Recv()` call we must return the same data, thus we remember it here.
+     */
+    mutable std::optional<uint8_t> m_peek_data;
```

Comment:
Either this should be `mutable` or the `const` qualifier should be removed from `Sock::Recv()` (and `Sock::Send()` for consistency). I don't have a strong opinion. I chose `mutable` because it is a smaller change and did not require modifying existing code.

Reply:
Removing `const` is not feasible (an avalanche of changes, not worth it).